oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/OAuth2IntrospectionClaimAccessor.java

```diff
@@ -34,107 +32,7 @@
-	 * @return the scopes associated with the token
-	 */
-	default String getScope() {
-		return this.getClaimAsString(OAuth2IntrospectionClaimNames.SCOPE);
-	}
-
-	/**
-	 * Returns the client identifier {@code (client_id)} for the token
-	 * @return the client identifier for the token
-	 */
-	default String getClientId() {
-		return this.getClaimAsString(OAuth2IntrospectionClaimNames.CLIENT_ID);
-	}
-
-	/**
-	 * Returns a human-readable identifier {@code (username)} for the resource owner that
-	 * authorized the token
-	 * @return a human-readable identifier for the resource owner that authorized the
-	 * token
-	 */
-	default String getUsername() {
-		return this.getClaimAsString(OAuth2IntrospectionClaimNames.USERNAME);
-	}
-
-	/**
-	 * Returns the type of the token {@code (token_type)}, for example {@code bearer}.
-	 * @return the type of the token, for example {@code bearer}.
-	 */
-	default String getTokenType() {
-		return this.getClaimAsString(OAuth2IntrospectionClaimNames.TOKEN_TYPE);
-	}
-
-	/**
-	 * Returns a timestamp {@code (exp)} indicating when the token expires
-	 * @return a timestamp indicating when the token expires
-	 */
-	default Instant getExpiresAt() {
-		return this.getClaimAsInstant(OAuth2IntrospectionClaimNames.EXPIRES_AT);
-	}
-
-	/**
-	 * Returns a timestamp {@code (iat)} indicating when the token was issued
-	 * @return a timestamp indicating when the token was issued
-	 */
-	default Instant getIssuedAt() {
-		return this.getClaimAsInstant(OAuth2IntrospectionClaimNames.ISSUED_AT);
-	}
-
-	/**
-	 * Returns a timestamp {@code (nbf)} indicating when the token is not to be used
-	 * before
-	 * @return a timestamp indicating when the token is not to be used before
-	 */
-	default Instant getNotBefore() {
-		return this.getClaimAsInstant(OAuth2IntrospectionClaimNames.NOT_BEFORE);
-	}
-
-	/**
-	 * Returns usually a machine-readable identifier {@code (sub)} of the resource owner
-	 * who authorized the token
-	 * @return usually a machine-readable identifier of the resource owner who authorized
-	 * the token
-	 */
-	default String getSubject() {
-		return this.getClaimAsString(OAuth2IntrospectionClaimNames.SUBJECT);
-	}
-
-	/**
-	 * Returns the intended audience {@code (aud)} for the token
-	 * @return the intended audience for the token
-	 */
-	default List<String> getAudience() {
-		return this.getClaimAsStringList(OAuth2IntrospectionClaimNames.AUDIENCE);
-	}
-
-	/**
-	 * Returns the issuer {@code (iss)} of the token
-	 * @return the issuer of the token
-	 */
-	default URL getIssuer() {
-		return this.getClaimAsURL(OAuth2IntrospectionClaimNames.ISSUER);
-	}
-
-	/**
-	 * Returns the identifier {@code (jti)} for the token
-	 * @return the identifier for the token
-	 */
-	default String getId() {
-		return this.getClaimAsString(OAuth2IntrospectionClaimNames.JTI);
-	}
+@Deprecated
+public interface OAuth2IntrospectionClaimAccessor extends OAuth2TokenIntrospectionClaimAccessor {
```

Comment:
We should make sure there are no more references to `OAuth2IntrospectionClaimAccessor` - `OAuth2IntrospectionAuthenticatedPrincipal` references it.

Reply:
Is it ok to break backward compatibility if we remove reference from OAuth2IntrospectionAuthenticatedPrincipal. Do we need OAuth2IntrospectionClaimAccessor then?
I thought OAuth2IntrospectionClaimAccessor will be removed in next major version.

Reply:
Sorry please ignore my comment as this would break backward compatibility.